### PR TITLE
Render system refactor

### PR DESCRIPTION
**Pull request by averhagen**
Refactor render system to delegate determination of conditional draw data out to smaller methods. This has some processing overhead but I think its worth it for the sake of keeping the code more neat and debuggable.

### CODE REVIEW

File: core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java [hunk 1]
```diff
@@ -26,10 +25,10 @@
     private ComponentMapper<Layer> layerComponentMapper;
     private ComponentMapper<Text> textComponentMapper;
 
-    private Text entityText;
     private Spatial entitySpatial;
     private Rotatable entityRotatable;
     private Renderable entityRenderable;
+    private Text entityText;
```
Comment (kevinfulkerson) [inline core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java:31]: Just out of curiosity, why did this move?

Comment (averhagen) [inline core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java:31]: Not sure.

File: core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java [hunk 2]
```diff
@@ -123,6 +113,47 @@ protected void processSystem() {
...
     }
 
+    private void drawEntityText(int entityId) {
+        if (textComponentMapper.has(entityId)) {
+            entityText = textComponentMapper.get(entityId);
+            entityText.font.draw(batch, entityText.text, entitySpatial.x, entitySpatial.y);
+        }
+    }
+
+    private void setEntityGraphics() {
```
Comment (kevinfulkerson) [inline core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java:126]: The call order is reversed here relative to before, does that matter at all? (see statements below)

File: core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java [hunk 3]
```diff
@@ -38,10 +37,20 @@
...
+    private float drawY;
+    private float drawOriginX;
+    private float drawOriginY;
+    private float drawWidth;
+    private float drawHeight;
+    private float drawRotation;
+    private boolean drawCurrentEntity;
+
     public RenderSystem(Camera camera, Aspect.Builder aspect, Texture missingTextureImage) {
         super(aspect.all(Layer.class).one(Text.class, Renderable.class));
```
Comment (kevinfulkerson) [inline core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java]: Should we consider adding Spatial to this aspect? All of these entities seem to require it to be drawn, and I think they all have it already, but it seems like it should be here.

Comment (averhagen) [inline core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java]: Yea I was going to add that but forgot. Adding that now.

File: core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java (hunk 2 above)
Comment (averhagen) [inline core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java:126]: Yea text should be drawn after if the entity has both. Changing that back now.

**Comment by averhagen** [comment]
Added some updates to this pull request based on the feedback.

**Comment by kevinfulkerson** [comment]
Looks good, merging.

**Pull request merged**